File: src/contests/cg/puzzle/csharp/teds-compiler.cs
// 🎮 CodinGame Puzzle - teds-compiler
// https://www.codingame.com/training/easy/teds-compiler

using System;

class Solution
{
    static void Main()
    {
        string line = Console.ReadLine().TrimEnd();
        int balance = 0, best = 0;
        for (int i = 0; i < line.Length; i++)
        {
            if (line[i] == '<') balance++; else balance--;
            if (balance < 0) break;
            if (balance == 0) best = i + 1;
        }
        Console.WriteLine(best);
    }
}

File: src/contests/cg/puzzle/java/teds-compiler.java
// 🎮 CodinGame Puzzle - teds-compiler
// https://www.codingame.com/training/easy/teds-compiler

import java.io.*;

class Solution {
    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        String line = br.readLine().replaceAll("\\s+$", "");
        int balance = 0, best = 0;
        for (int i = 0; i < line.length(); i++) {
            if (line.charAt(i) == '<') balance++;
            else balance--;
            if (balance < 0) break;
            if (balance == 0) best = i + 1;
        }
        System.out.println(best);
    }
}
